### DISCUSSION TITLE
ZIgbee2mqtt adapters

### DISCUSSION THREAD

**Discussion by sercio** [Koenkk/zigbee2mqtt#19491] [General]
Hi,
in aliexpress there are a lot of adatpter with zigbee2mqtt indication (if you search cc2652 zigbee), also it appera with different chip.
Do you know if they are compatible with supervisor firmware?
Thanks

**Comment by francisp2** [comment]
> Do you know if they are compatible with supervisor firmware?

I think you want to say 'coordinator firmware' ?

**Comment by sercio** [comment]
Yes, sorry Coordinator.
On the adapters you can see "Zigbee2mqtt Dongle Plus smarthome stick" and others as "CH9102X" or similar...
Thanks

________________________________
Da: francisp2 ***@***.***>
Inviato: martedì 31 ottobre 2023 07:01
A: Koenkk/zigbee2mqtt ***@***.***>
Cc: sercio ***@***.***>; Author ***@***.***>
Oggetto: Re: [Koenkk/zigbee2mqtt] ZIgbee2mqtt adapters (Discussion #19491)


Do you know if they are compatible with supervisor firmware?
I think you want to say 'coordinator firmware' ?

**Reply by sercio** [reply]
Are sponsorized by ZIgbee2mqtt?

**Reply by DubhAd** [reply]
No, they're not

**Comment by jeroenvermeulen** [comment]
I have one from [AliExpress](https://nl.aliexpress.com/item/1005006155691159.html).
Only the stick, I already had a CC-Debugger laying around.
I connected the CC-Debugger, the small cable can only go one way with the housing on the stick, which is the same direction as it goes on the CC2531.
However: **When I connected both the CC-Debugger and the stick to USB I smelt some electronics burning**.
I quickly disconnected and tried again using my lab power supply with a over-current-protection of 0.5 A.
It is the "Zigbee2mqtt Dongle Plus Smart Home Stick CH9102X" stick which is drawing too much current.
Of course I had the red line of the small and langer flatcables aligned to the 1 pins.

**Comment by mrneutron42** [comment]
I recently ordered and tried loading the Z-stack Zibgee 3.0 coordinator firmware into a Zigbee2mqtt Dongle Plus Smart Home Stick. 
It’s got the high power CC2652P Zigbee chip and the Silicon Labs CP2102N USB-to-serial interface chip in it.
see:
(https://www.amazon.com/dp/B0C7GMVQKR)

(https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin)

I downloaded the TI Flash Programmer 2 software and tried to load the Z-Stack firmware into it via the USB-serial interface. I successfully loaded the Windows drivers for the Silicon Labs CP2102N chip and Windows recognized it as a serial port on COM3.

The Flash Programmer 2 software DOES allow me to read the MAC address and read the Info Page data, after the RESET + K1 switch combination.
But, it would not go into Bootloader mode and load new firmware after the the RESET + K1 switch sequence (which is supposed to put the CC2652P into bootloader mode).

The seller’s instruction for bootloader mode is:
“after connecting the USB, press and hold the RESET button, press and hold KEY1, then release the RESET, and then release the KEY1 button, then the device will enter the BOOTLOAD mode.”

I eventually gave up and returned it to Amazon during my product return window.

**Comment by sercio** [comment]
Have you tryed with the python method?
Instruction at this link
https://www.youtube.com/watch?v=JM9LnwtZRoA

________________________________
Da: mrneutron42 ***@***.***>
Inviato: mercoledì 6 dicembre 2023 17:58
A: Koenkk/zigbee2mqtt ***@***.***>
Cc: sercio ***@***.***>; Author ***@***.***>
Oggetto: Re: [Koenkk/zigbee2mqtt] ZIgbee2mqtt adapters (Discussion #19491)


I recently ordered and tried loading the Z-stack Zibgee 3.0 coordinator firmware into a Zigbee2mqtt Dongle Plus Smart Home Stick.
It’s got the high power CC2652P Zigbee chip and the Silicon Labs CP2102N USB-to-serial interface chip in it.
see:
(https://www.amazon.com/dp/B0C7GMVQKR)

(https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin)

I downloaded the TI Flash Programmer 2 software and tried to load the Z-Stack firmware into it via the USB-serial interface. I successfully loaded the Windows drivers for the Silicon Labs CP2102N chip and Windows recognized it as a serial port on COM3.

The Flash Programmer 2 software DOES allow me to read the MAC address and read the Info Page data, after the RESET + K1 switch combination.
But, it would not go into Bootloader mode and load new firmware after the the RESET + K1 switch sequence (which is supposed to put the CC2652P into bootloader mode).

The seller’s instruction for bootloader mode is:
“after connecting the USB, press and hold the RESET button, press and hold KEY1, then release the RESET, and then release the KEY1 button, then the device will enter the BOOTLOAD mode.”

I eventually gave up and returned it to Amazon during my product return window.

**Reply by mrneutron42** [reply]
No. I didn't try the Python method, and I returned the dongle for a refund.

**Reply by jeroenvermeulen** [reply]
@sercio Thank you I will try within a couple of days.
To be continued...

**Comment by jeroenvermeulen** [comment]
Big thanks to @sercio for pointing me in the right direction.

Reminder for those who are not reading the whole thread: 
**Do NOT connect the CC Debugger to this stick**

I was able to get the ["Zigbee2mqtt Dongle Plus Smart Home Stick CH9102X"](https://www.aliexpress.com/item/1005006155691159.html) into bootloader mode by:
- removing the case
- holding the K1 button (the middle one)
- pressing and releasing the RST button (the one on the side of the antenna)
- releasing the K1 button

Next I was able to flash using these commands (on macOS):
```
git clone https://github.com/JelmerT/cc2538-bsl.git
cd cc2652p-bsl
wget https://github.com/Koenkk/Z-Stack-firmware/raw/master/coordinator/Z-Stack_3.x.0/bin/CC1352P2_CC2652P_launchpad_coordinator_20230507.zip
unzip CC1352P2_CC2652P_launchpad_coordinator_20230507.zip
pip install wheel pyserial intelhex python-magic
pip install zigpy-znp
python3 -m zigpy_znp.tools.nvram_read /dev/tty.usbserial-* -o nvram_backup.json
python3 cc2538-bsl.py -e -v -w  CC1352P2_CC2652P_launchpad_coordinator_20230507.hex
```

And **it works**!

**Reply by mrneutron42** [reply]
Can you try it with the TI Flash Programmer 2 software?

I believe you need to put the dongle into bootloader mode with the same K1 and RESET button sequence, and then you can use the TI Flash Programmer 2.
https://electrolama.com/radio-docs/flash-ti-flash-prog/

The dongle I had would not go into bootloader mode so I was dead in the water.

**Reply by jeroenvermeulen** [reply]
OS: Windows 10 Home 22H2
CPU: Intel Core i7
Firmware: CC1352P2_CC2652P_launchpad_coordinator_20230507.hex
Programmer software: SmartRF Flash Programmer v2 version 1.8.2 (05 Jul 2020)
USB Device: USB-Enhanced Serial CH9102 (COM4)
Target: CC2652P
[x] Erase  => works
[x] Program => not working. Error `Unknown record type 3`
https://github.com/Koenkk/Z-Stack-firmware/issues/397

**Reply by mrneutron42** [reply]
Huh.  Maybe the one I had would have worked with the Python method?

**Reply by jeroenvermeulen** [reply]
Yes, that is possible.

I can indeed read the mac and other data from the stick using Flash Programmer v2, when I put the stick in Bootloader mode.
Before each read action I have to put it in Bootloader mode it again.
I tried with UniFlash but the stick is not auto-detected and this software has many options for manual selection. Did not find a working combination.

The Python method is the only thing working flawless.

**Reply by jeroenvermeulen** [reply]
But I won't recommend to buy the "Zigbee2mqtt Dongle Plus Smart Home Stick CH9102X" again.
I would pick one from this list https://www.zigbee2mqtt.io/guide/adapters/
Most people use these and there are instructions available.

**Reply by mrneutron42** [reply]
Sounds like you had the same experience
With the TI Flash Programmer 2 software, I was also able to read the dongle's Zigbee MAC address, but could not program.

I ordered a Sonoff after returning the zigbee2mqtt dongle plus.
https://itead.cc/product/sonoff-zigbee-3-0-usb-dongle-plus/

**Reply by nitr0man** [reply]
I had CH9102X dongle, it's flashed OK, but there's troubles with device registration on last z-stack firmware https://github.com/Koenkk/zigbee2mqtt/issues/20317

I hope it can be solved, but i've already ordered different dongle on efr32.

**Reply by nitr0man** [reply]
update: it's paired and work with CC1352P2_CC2652P_launchpad_coordinator_20221226. after upgrade to latest with nvram reset before z2m start - it looks to work OK too.

**Reply by nitr0man** [reply]
update 2: zigbee devices are lost after some time (from some minutes till teens of hours). maybe trouble is in serial driver (I use cdc instead of ch343).

**Reply by marcosmoura91** [reply]
I just want to report that I also was able to flash to my aliexpress CC2652P module. 

I did it while connected to raspberry pi and i had to run as follows: 

```
git clone https://github.com/JelmerT/cc2538-bsl.git
cd cc2652p-bsl/cc2652p_bsl
wget https://github.com/Koenkk/Z-Stack-firmware/releases/download/Z-Stack_3.x.0_coordinator_20250321/CC1352P2_CC2652P_launchpad_coordinator_20250321.zip
unzip CC1352P2_CC2652P_launchpad_coordinator_20230507.zip
sudo pip3 install wheel pyserial intelhex python-magic
sudo pip3 install zigpy-znp

sudo python3 -m zigpy_znp.tools.nvram_read /dev/ttyACM0 -o nvram_backup.json

#now to write we need to go into bootloader mode. Press Key1 hold, reset hold, release reset and after that release key1.
sudo python3 cc2538_bsl.py -e -v -w  CC1352P2_CC2652P_launchpad_coordinator_20250321.hex
```

**Reply by trenchcoatjedi** [reply]
> Big thanks to @sercio for pointing me in the right direction.
> 
> 
> 
> Reminder for those who are not reading the whole thread: 
> 
> **Do NOT connect the CC Debugger to this stick**
> 
> 
> 
> I was able to get the ["Zigbee2mqtt Dongle Plus Smart Home Stick CH9102X"](https://www.aliexpress.com/item/1005006155691159.html) into bootloader mode by:
> 
> - removing the case
> 
> - holding the K1 button (the middle one)
> 
> - pressing and releasing the RST button (the one on the side of the antenna)
> 
> - releasing the K1 button
> 
> 
This method worked to flash with SMLIGHT online flasher also

**Comment by gyrex** [comment]
I've bought this dongle and successfully flashed the router firmware (CC1352P2_CC2652P_launchpad_router_20250403.hex) since I'm already using a Sonoff Zigbee P as the main coordinator.

I initially used the python script but as @trenchcoatjedi mentioned above, the SMLIGHT online flasher is probably the easiest method (https://smlight.tech/flasher/#other_cc)

My issue is that I'm trying to figure out how to put the dongle in pairing mode so I can add it  to my existing zigbee network. I've tried long pressing on all buttons to no avail. Any ideas?
